pkg/wekafs/dirvolume.go

```diff
@@ -0,0 +1,388 @@
+package wekafs
+
+import (
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/google/uuid"
+	"github.com/pkg/xattr"
+	"github.com/wekafs/csi-wekafs/pkg/wekafs/apiclient"
+	"golang.org/x/sys/unix"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+)
+
+type DirVolume struct {
+	id         string
+	Filesystem string
+	volumeType string
+	dirName    string
+	apiClient  *apiclient.ApiClient
+}
+
+var ErrNoXattrOnVolume = errors.New("xattr not set on volume")
+var ErrBadXattrOnVolume = errors.New("could not parse xattr on volume")
+
+func (v DirVolume) getMaxCapacity(mountPath string) (int64, error) {
+	glog.V(5).Infoln("Attempting to get max capacity available on filesystem", v.Filesystem)
+	var stat syscall.Statfs_t
+	err := syscall.Statfs(mountPath, &stat)
+	if err != nil {
+		return -1, status.Errorf(codes.FailedPrecondition, "Could not obtain free capacity on mount path %s: %s", mountPath, err.Error())
+	}
+	// Available blocks * size per block = available space in bytes
+	maxCapacity := int64(stat.Bavail * uint64(stat.Bsize))
+	glog.V(4).Infoln("Max capacity available for a volume:", maxCapacity)
+	return maxCapacity, nil
+}
+
+func (v DirVolume) GetType() VolumeType {
+	return VolumeTypeDirV1
+}
+
+func (v DirVolume) GetCapacity(mountPath string) (int64, error) {
```

Comment:
- handle error from getSizeFromQuota. On 500 and 404 fall back to xattr (no need for df)
- if no client - use xattr
- why do we need df? can't come up with scenario